src/Fields.jl

```diff
@@ -42,3 +42,8 @@ function factor(x::FieldElement)
   @req !is_zero(x) "Element must be non-zero"
   return Fac(x, Dict{typeof(x), Int}())
 end
+
+function canonical_unit(x::FieldElement)
+  iszero(x) && return one(x)
```

Comment:
Also pushed this and a few similar chjanges to get us a step closer to finally resolving issue #1066 -- there may still be `canonical_unit` methods which may return non-units, but at least this change means there are a bunch fewer. 

CC @thofma @fieker

Reply:
Thanks, LGTM! For the cases where a `iszero(x)` call is either extremely expensive or impossible (I am thinking of Arb/Acb constructions), IMO the particular types themselves should then overload `canonical_unit` with an appropriate implementation